File: NFCInHome_Server/src/nfcinhome_server/Translator.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

package nfcinhome_server;

import java.util.ArrayList;

/**
 *
 * @author fhict
 */
public class Translator {
    
    private final ArrayList<command> translations;

    public Translator(){
        translations = new ArrayList<>();
        fillTranslations();
    }

    private void fillTranslations() {
        translations.add(new command("livingroom_table_lamp", "Table Lamp", 11));
        translations.add(new command("livingroom_couch_lamo", "Couch Lamp", 12));
    }
    
    public String getCommandName(int command){
        for (command c : translations){
            if (c.getCommando() == command){
                return c.getName();
            }
        }
        return null;
    }
}

File: NFCInHome_Server/NFCInHome_Server/Translator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace NFCInHome_Server
{
    class Translator
    {
        private List<Command> translations;

        public Translator()
        {
            translations = new List<Command>();
            fillTranslations();
        }

        private void fillTranslations()
        {
            translations.Add(new Command("livingroom_table_lamp", "Table Lamp", 11, false, NFCInHome_Server.Properties.Resources.livingroom_table_lamp_on, NFCInHome_Server.Properties.Resources.livingroom_table_lamp_off));
            translations.Add(new Command("livingroom_couch_lamp", "Couch Lamp", 12, false, NFCInHome_Server.Properties.Resources.livingroom_couch_lamp_on, NFCInHome_Server.Properties.Resources.livingroom_couch_lamp_off));
        }

        public Command getCommand(int command)
        {
            foreach (Command c in translations)
            {
                if (c.getCommand() == command)
                {
                    return c;
                }
            }
            return null;
        }
    }
}
